src/App.tsx

```diff
@@ -7,33 +8,53 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
 
-        <span className="Clock__time">
-          {date.toLocaleTimeString()}
-        </span>
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  componentDidMount() {
+    window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('click', () => {
+      this.setState({
+        hasClock: true,
+      });
+    });
+    document.addEventListener('contextmenu', () => {
+      this.setState({
+        hasClock: false,
+      });
+    });
+  }
+
+  componentDidUpdate(_: Readonly<{}>, prevState: State) {
+    const prevName = prevState.clockName;
+    const newName = this.state.clockName;
+
+    if (prevName !== newName) {
+      // eslint-disable-next-line
+      console.log(`Renamed from ${prevName} to ${newName}`);
+    }
```

Comment:
Find the way to not disable linter) Think of what particular element of DOM is using console.log methods)